### PR TITLE
QSysify emitters for AXI4, AXI4 Lite and Interrupt Sender

### PR DESCRIPTION
**Pull request by c-thaler**
Configs for AXI4 and AXI4 had to be extended:
max number of pending reads/writes and reorder FIFO depth

I also added a check for AXI4 Lite data width. The spec only allows 32 and 64 bit.

### CODE REVIEW

File: lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4.scala [hunk 1]
```diff
@@ -49,11 +49,16 @@ case class Axi4Config(addressWidth : Int,
...
-                      arUserWidth   : Int = -1,
-                      awUserWidth   : Int = -1,
+                      arUserWidth  : Int = -1,
+                      awUserWidth  : Int = -1,
                       rUserWidth   : Int = -1,
                       wUserWidth   : Int = -1,
-                      bUserWidth   : Int = -1) {
+                      bUserWidth   : Int = -1,
+                      
+                      readIssuingCapability     : Int = 1,
```
Comment (Dolu1990) [inline lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4.scala:58]: We might want to put it to Int.MaxValue or -1 by default ? to say "unspecified" by default ?

**Comment by Dolu1990** [comment]
Sorry for the delay :)
see my above review comment about AXI.

File: lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4.scala (hunk 1 above)
Comment (c-thaler) [reply lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4.scala:58]: Yep, makes sense. We should use -1 by default and prevent the QSysifyEmitter from emitting the corresponding `set_interface_property` in this case.
I think that the `*Capability` parameters are optional, so Quartus will use some "safe" default. I'm not sure of this, I will check it ASAP.

Comment (c-thaler) [reply lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4.scala:58]: Quartus Platform Designer defaults to 1 when `*Capability` is not set in the TCL script. I will proceed as proposed in my former comment.

**Comment by c-thaler** [comment]
The QSys parameters are now disabled by default.

**Comment by Dolu1990** [comment]
ok :)
So all good for merge ?

**Comment by c-thaler** [comment]
Yes, please :)

**Comment by Dolu1990** [comment]
Thanks ^^

**Pull request merged**